internal/connection/spinner.go

```diff
@@ -0,0 +1,94 @@
+package connection
+
+import (
+	"fmt"
+
+	"github.com/charmbracelet/bubbles/spinner"
+	tea "github.com/charmbracelet/bubbletea"
+	"github.com/charmbracelet/lipgloss"
+)
+
+const spinnerText = `Connecting to the cluster %s at %s.
+   Check the logs at %s.`
+
+var (
+	currSpinner  = spinner.Dot
+	spinnerStyle = lipgloss.NewStyle().Foreground(lipgloss.Color("#40826D"))
+	helpStyle    = lipgloss.NewStyle().Foreground(lipgloss.Color("241")).Render
+)
+
+type connectionSpinnerModel struct {
+	clusterName string
+	address     string
+	logFileName string
+	spinner     spinner.Model
+	escaped     *bool
+}
+
+func newConnectionSpinnerModel(clusterName, address, filename string, escaped *bool) *connectionSpinnerModel {
+	m := &connectionSpinnerModel{}
+	m.spinner = spinner.New()
+	m.spinner.Style = spinnerStyle
+	m.spinner.Spinner = currSpinner
+	m.clusterName = clusterName
+	m.address = address
+	m.logFileName = filename
+	m.escaped = escaped
+	return m
+}
+
+func (m connectionSpinnerModel) Init() tea.Cmd {
+	return m.spinner.Tick
+}
+
+type Quitting struct {
+}
+
+type EmptyDisplay struct {
+	quit bool
+}
+
+func (e *EmptyDisplay) Init() tea.Cmd {
+	return nil
+}
+
+func (e *EmptyDisplay) Update(_ tea.Msg) (tea.Model, tea.Cmd) {
+	if e.quit {
+		return e, tea.Quit
+	}
+	return e, nil
+}
+
+func (e *EmptyDisplay) View() string {
+	e.quit = true
+	return ""
+}
+
+func (m connectionSpinnerModel) Update(msg tea.Msg) (tea.Model, tea.Cmd) {
```

Comment:
This should be a pointer receiver, since you are modifying it.

Reply:
If you imply `escaped` variable by modifying. It is a pointer so there is need for setting method as a pointer receiver. If you imply `m.spinner, cmd = m.spinner.Update(msg)` this statement, the method also return `tea.Model` while updating. It is the general usage of spinner (recommended way). It looks becoming pointer receiver or value receiver is not matter for this method.